### PR TITLE
Generate supported-backends markdown table

### PR DESCRIPTION
**Pull request by kaizhong2021**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: requirements/tests.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 asynctest
 coverage
+easydict
```
Comment (RunningLeon) [inline requirements/tests.txt:3]: Could remove this package.

File: tools/generate_md_table.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import argparse
+import os
+import os.path as osp
+
+import yaml
+from easydict import EasyDict as edict
```
Comment (RunningLeon) [inline tools/generate_md_table.py:7]: could remove

File: tools/generate_md_table.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    header = ['model', 'task'] + args.backends
+    aligner = [':--'] * 2 + [':--'] * len(args.backends)
+
+    def write_row_f(writer, row):
+        writer.write('|' + '|'.join(row) + '|\n')
+
+    print(f'Processing{args.yml_file}')
+    with open(args.yml_file, 'r') as reader, open(args.output, 'w') as writer:
+        config = yaml.load(reader, Loader=yaml.FullLoader)
+        config = edict(config)
```
Comment (RunningLeon) [inline tools/generate_md_table.py:44]: config = Config(config)

File: tests/regression/mmseg.yml [hunk 4]
```diff
@@ -1,4 +1,5 @@
 globals:
+  repo_url: https://github.com/open-mmlab/mmsegmentation/tree/master
```
Comment (RunningLeon) [inline tests/regression/mmseg.yml:2]: use `main` branch: `https://github.com/open-mmlab/mmsegmentation/tree/main`

File: tests/regression/mmpose.yml [hunk 5]
```diff
@@ -1,4 +1,5 @@
 globals:
+  repo_url: https://github.com/open-mmlab/mmpose/tree/master
```
Comment (RunningLeon) [inline tests/regression/mmpose.yml:2]: use `main` branch

File: tests/regression/mmedit.yml [hunk 6]
```diff
@@ -1,4 +1,5 @@
 globals:
+  repo_url: https://github.com/open-mmlab/mmediting/tree/master
```
Comment (RunningLeon) [inline tests/regression/mmedit.yml:2]: use `main` branch

File: tests/regression/mmdet.yml [hunk 7]
```diff
@@ -1,4 +1,5 @@
 globals:
+  repo_url: https://github.com/open-mmlab/mmdetection/tree/master
```
Comment (RunningLeon) [inline tests/regression/mmdet.yml:2]: use `main` branch

File: tests/regression/mmcls.yml [hunk 8]
```diff
@@ -1,4 +1,5 @@
 globals:
+  repo_url: https://github.com/open-mmlab/mmclassification/tree/master
```
Comment (RunningLeon) [inline tests/regression/mmcls.yml:2]: use mmpretrain `main` branch : https://github.com/open-mmlab/mmpretrain/tree/main

File: tests/regression/mmdet3d.yml [hunk 9]
```diff
@@ -1,4 +1,5 @@
 globals:
+  repo_url: https://github.com/open-mmlab/mmdetection3d/tree/master
```
Comment (RunningLeon) [inline tests/regression/mmdet3d.yml:2]: use `main` branch

File: tests/regression/mmaction.yml [hunk 10]
```diff
@@ -1,4 +1,5 @@
 globals:
+  repo_url: https://github.com/open-mmlab/mmaction2/tree/master
```
Comment (RunningLeon) [inline tests/regression/mmaction.yml:2]: use `https://github.com/open-mmlab/mmaction2/tree/main`

**Comment by RunningLeon** [comment]
@kaizhong2021 Could update as comments suggest.

**Comment by kaizhong2021** [comment]
> @kaizhong2021 Could update as comments suggest.
Ok , teacher I will update it later

File: tests/regression/mmcls.yml [hunk 11]
```diff
@@ -1,4 +1,5 @@
 globals:
+  repo_url: https://github.com/open-mmlab/mmpretrain/tree/main
   codebase_dir: ../mmclassification
```
Comment (RunningLeon) [inline tests/regression/mmcls.yml:3]: codebase_dir: `../mmpretrain`

File: tools/generate_md_table.py [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+from mmdeploy.utils import get_backend, get_task_type, load_config
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='from yaml export markdown table')
+    parser.add_argument('yml_file', help='input yml config path')
+    parser.add_argument('output', help='output markdown file path')
+    parser.add_argument(
+        'backends',
```
Comment (RunningLeon) [inline tools/generate_md_table.py:18]: --backends, nargs='+'

File: docs/zh_cn/02-how-to-run/useful_tools.md [hunk 13]
```diff
@@ -202,3 +202,38 @@ python tools/profiler.py \
...
+    ${yml_file} \
+    ${output} \
+    ${backends}
+```
+
+### 参数说明
+
+- `yml_file:`  输入 yml 配置路径
+- `output:` 输出markdown文件路径
+- `backends:` 要输出的后端，默认为 onnxruntime tensorrt torchscript pplnn openvino ncnn
```
Comment (RunningLeon) [inline docs/zh_cn/02-how-to-run/useful_tools.md:223]: `--backends`

File: docs/zh_cn/02-how-to-run/useful_tools.md [hunk 14]
```diff
@@ -202,3 +202,38 @@ python tools/profiler.py \
...
+
+### 使用举例
+
+从 mmocr.yml 生成mmdeploy支持的后端表
+
+```shell
+python tools/generate_md_table.py tests/regression/mmocr.yml tests/regression/mmocr.md onnxruntime tensorrt torchscript pplnn openvino ncnn
```
Comment (RunningLeon) [inline docs/zh_cn/02-how-to-run/useful_tools.md:230]: `--backends`

File: tools/generate_md_table.py [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+
+
+def main():
+    args = parse_args()
+    assert osp.exists(args.yml_file), f'File not exists: {args.yml_file}'
+    output_dir, _ = osp.split(args.output)
+    if output_dir:
+        os.makedirs(output_dir, exist_ok=True)
+    header = ['model', 'task'] + args.backends
+    aligner = [':--'] * 2 + [':--'] * len(args.backends)
```
Comment (RunningLeon) [inline tools/generate_md_table.py:36]: @kaizhong2021 hi, could make center alignment starting from coloumn 3rd.
```
aligner = [':--'] * 2 + [':-:'] * len(args.backends)
```

Comment (kaizhong2021) [reply tools/generate_md_table.py:36]: OK，teacher，I will fix it later

**Comment by RunningLeon** [comment]
@kaizhong2021 hi, could you kindly fix bad urls found in lint errors: such as https://github.com/open-mmlab/mmdeploy/actions/runs/4716951096/jobs/8365124371?pr=1986#step:10:436

**Comment by kaizhong2021** [comment]
> @kaizhong2021 hi, could you kindly fix bad urls found in lint errors: such as https://github.com/open-mmlab/mmdeploy/actions/runs/4716951096/jobs/8365124371?pr=1986#step:10:436

OK，Teacher I will fix it later

**Review by irexyc — APPROVED** [review]

**Pull request merged**